### ISSUE TITLE
RegisterEvidences(cdc) currently registers mock evidence types

### ISSUE THREAD

**Issue by ValarDragon** [tendermint/tendermint#2525]
Currently register evidences is as follows:
```
func RegisterEvidences(cdc *amino.Codec) {
	cdc.RegisterInterface((*Evidence)(nil), nil)
	cdc.RegisterConcrete(&DuplicateVoteEvidence{}, "tendermint/DuplicateVoteEvidence", nil)

	// mocks
	cdc.RegisterConcrete(MockGoodEvidence{}, "tendermint/MockGoodEvidence", nil)
	cdc.RegisterConcrete(MockBadEvidence{}, "tendermint/MockBadEvidence", nil)
}
```
I think this will lead to unexpected behavior happening if people start submitting MockGoodEvidence, since we're currently supporting that. We should separate this into `RegisterEvidence` and `RegisterMockEvidence`.

**Comment by ValarDragon** [comment]
Not sure whats the best way to solve this is. I think we'd have to make more methods in types take a codec instead of having it implicit in the package. For example, `MakePartSet` needs a codec in order to distinguish between package default and the mock testing codec.

**Comment by melekes** [comment]
@ValarDragon what if we expose `GetCodec()` method and use it to inject mock types during a test https://github.com/tendermint/tendermint/pull/2571

**Comment by melekes** [comment]
I don't like the current per package code approach, but I guess we can live with that in the short term.

**Comment by ValarDragon** [comment]
closed by #2571
